src/services/CertificateGenerationService.ts

```diff
@@ -500,11 +500,30 @@ class CertificateGenerationService {
           // Remove the first result as it should be the current one.
           testResults.shift();
 
+          // Remove any failed test results as they should not to be included in Odometer History
+          let filteredTestResults: any[] = [];
+
+          testResults.forEach((testResult) => {
+            let hasPassOrPrs: boolean = false;
+            if (testResult.testTypes) {
+              testResult.testTypes.forEach((testType: ITestType) => {
+                if (testType.testResult !== "fail") {
+                  hasPassOrPrs = true;
+                }
+              });
+              if (hasPassOrPrs) {
+                filteredTestResults.push(testResult);
+              }
+            } else {
+              filteredTestResults.push(testResult);
+            }
+          });
```

Comment:
Can we discuss this logic. I think it can be simplified by using `filter` instead of `foreach`